### PR TITLE
feat: add RequestListView

### PR DESCRIPTION
**Pull request by gca-axelor**
RM#86192

### CODE REVIEW

File: packages/apps/purchase/src/api/supplier-api.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+  return criterias;
+};
+
+export async function searchSupplier({page = 0, searchValue}) {
+  return createStandardSearch({
+    model: 'com.axelor.apps.base.db.Partner',
+    criteria: createSupplierCriteria(searchValue),
+    fieldKey: 'purchase_supplier',
+    sortKey: 'purchase_supplier',
```
Comment (vhu-axelor) [inline packages/apps/purchase/src/api/supplier-api.js:47]: Declare it or remove it

File: packages/apps/purchase/src/models/typeObjects.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+          {
+            key: 'Requested',
+            value: 2,
+            title: 'Purchase_Status_Requested',
+            color: 'progressColor',
+          },
+          {
+            key: 'Accepted',
+            value: 3,
+            title: 'Purchase_Status_Planned',
```
Comment (vhu-axelor) [inline packages/apps/purchase/src/models/typeObjects.ts:42]: Change the title (and trads behind) to match the key

**Review by vhu-axelor — APPROVED** [review]

File: packages/apps/purchase/src/api/supplier-api.js (hunk 1 above)
Comment (lme-axelor) [reply packages/apps/purchase/src/api/supplier-api.js]: That's not true. As a good practice, those keys should be always present to enable client projects to override the value. If it's not present, then the client project will need to rewrite the whole API/slice only to modify sorting.

So if we don't need it in standard, then no need to declare it but it should always be put in the API call so some other package can easily declare it.

File: packages/apps/purchase/src/i18n/fr.json [hunk 3]
```diff
@@ -1,3 +1,13 @@
 {
-  "Purchase_Purchase": "Achats"
+  "Purchase_Purchase": "Achats",
+  "Purchase_InternalRequests": "Demandes internes",
+  "Purchase_Status_Draft": "Brouillon",
+  "Purchase_Status_Requested": "Demandé",
+  "Purchase_Status_Accepted": "Accepté",
+  "Purchase_Status_Purchased": "Acheté",
+  "Purchase_Status_Refused": "Refusé",
+  "Purchase_Status_Canceled": "Annulé",
```
Comment (lme-axelor) [inline packages/apps/purchase/src/i18n/fr.json:9]: I think we should be an ending 'e' on this statuses as "Demande" is a feminine name

File: packages/apps/purchase/src/models/sortFields.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import {SortFields} from '@axelor/aos-mobile-core';
+
+export const purchase_sortFields: SortFields = {
+  purchase_purchaseRequest: ['statusSelect'],
```
Comment (lme-axelor) [inline packages/apps/purchase/src/models/sortFields.ts:22]: sort them by purchaseRequestSeq too to get the latest request as the top for each status

File: packages/apps/purchase/src/models/typeObjects.ts [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+            key: 'Refused',
+            value: 5,
+            title: 'Purchase_Status_Refused',
+            color: 'cautionColor',
+          },
+          {
+            key: 'Canceled',
+            value: 6,
+            title: 'Purchase_Status_Canceled',
+            color: 'errorColor',
```
Comment (lme-axelor) [inline packages/apps/purchase/src/models/typeObjects.ts:61]: Inverse the color of those statuses

File: packages/apps/purchase/src/screens/RequestListScreen.tsx [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+
+  const [selectedStatus, setSelectedStatus] = useState([]);
+  const [supplier, setSupplier] = useState();
+
+  const statusList = useMemo(() => {
+    const statusToDisplay = [
+      PurchaseRequest?.statusSelect.Draft,
+      PurchaseRequest?.statusSelect.Requested,
+      PurchaseRequest?.statusSelect.Accepted,
+    ];
```
Comment (lme-axelor) [inline packages/apps/purchase/src/screens/RequestListScreen.tsx:53]: I think we should either show all statuses in the ChipSelect or only display the concerned statuses in the list.
Currently, it's not really consistent. Let's check with MCH when he comes back.

**Pull request merged**